scripts/mkstatic.py

```diff
@@ -291,6 +292,23 @@ class API(object):
    public :: {subroutines}
 
    contains
+
+   ! Necessary to convert incoming suite and group names to lowercase
+   function to_lower(str) result(lower)
+      implicit none
+      character(len=*), intent(in) :: str
+      character(len=len(str)) :: lower
+      integer :: i, ichar_val
+
+      do i = 1, len(str)
+         ichar_val = ichar(str(i:i))
+         if (ichar_val >= ichar('A') .and. ichar_val <= ichar('Z')) then
+            lower(i:i) = char(ichar_val + 32)
```

Comment:
I think this would be clearer (and adhere to the Fortran standard*) if there was a parameter, say upper_to_lower instead of a literal `32`:
```
integer, parameter :: upper_to_lower = ichar('a') - ichar('A')
```
```suggestion
            lower(i:i) = char(ichar_val + upper_to_lower)
```
* the Fortran collating sequence does not require lowercase letters to come after the uppercase letters, nor does it specify a distance. An alternative (less good IMHO) would be to use `iachar` and `achar` which at least guarantee working with ASCII values.

Reply:
That's a good idea. I will make the change and test it.

Reply:
Done in https://github.com/NCAR/ccpp-framework/pull/669/commits/0db01c9c7b6426ef5b8013b68d3028ef4818de65